runestone/activecode/js/activecode.js

```diff
@@ -1222,6 +1227,35 @@ Yet another is that there is an internal error.  The internal error message is:
         }
     }
 
+    async runCoaches() {
+        //Run all available code coaches and update code coach div
+
+        //clear anything after header in codecoach div and hide it
+        $(this.codecoach).children().slice(1).remove();
+        $(this.codecoach).css("display", "none");
+
+        //get code, run coaches
+        let code = await this.buildProg(false);
+        let results = [];
+        for(let coach of this.codeCoachList) {
+            results.push(coach.check(code));
```

Comment:
Should there be a method to add a coach?  If I'm thinking about the future and adding a new coach I would want to have a method to do that so I don't have to modify this code...

Reply:
No one should have to touch that code. 

Coaches are added up in the constructor (line 143). I'm not sure where else they would be injected from - that was one of my questions. I don't have a good sense of how to make creating coaches more modular.

Reply:
Hrm, can't figure out how to quote in those lines without starting a new conversation.

Reply:
I was imagining a method on the Activecode instance, but then I'm not sure what other object would call it, so having it in the constructor is fine for now.